lib/src/install.rs

```diff
@@ -751,21 +773,30 @@ async fn install_container(
 
     // Pull the container image into the target root filesystem. Since this is
     // an install path, we don't need to fsync() individual layers.
-    let pulled_image = {
-        let spec_imgref = ImageReference::from(src_imageref.clone());
-        let repo = &sysroot.repo();
-        repo.set_disable_fsync(true);
-        let r = crate::deploy::pull(
+    let spec_imgref = ImageReference::from(src_imageref.clone());
+    let repo = &sysroot.repo();
+    repo.set_disable_fsync(true);
+    let (prepared_image, existing_image) =
+        crate::deploy::prepare_for_pull(repo, &spec_imgref, Some(&state.target_imgref)).await?;
+
+    let pulled_image = if existing_image.is_some() {
+        existing_image.unwrap()
```

Comment:
This is better written as
```
    let pulled_image = if let Some(img) = existing_image {
        img
    } else {
      ....
}

Reply:
ah thanks, I just force pushed a cleanup of these conditionals. I think I get how this is supposed to work now, but the extra `else { anyhow::bail!}` seems off